eng/build.ps1

```diff
@@ -115,9 +155,69 @@ function Build-Repo() {
     /p:Execute=$execute `
     /p:ContinuousIntegrationBuild=$ci `
     /p:VisualStudioDropName=$vsDropName `
+    /p:IbcOptimizationDataDir=$optDataDir `
+    /p:OfficialBuildId=$officialBuildId `
     @properties
 }
 
+function Restore-OptProfData() {
+    $dropToolDir = Get-PackageDir "Drop.App"
+    $dropToolPath = Join-Path $dropToolDir "lib\net45\drop.exe"
+
+    if (!(Test-Path $dropToolPath)) {
+
+        # Only report error when running in an official build.
+        # Allows to test optimization data operations locally by running
+        # cibuild.cmd after manually restoring internal tools project.
+        if (!$officialBuildId) {
+            $script:applyOptimizationData = $false
+            return
+        }
+
+        Write-Host "Internal tool not found: '$dropToolPath'." -ForegroundColor Red
+        Write-Host "Run nuget restore `"$EngRoot\internal\Toolset.csproj`"." -ForegroundColor DarkGray
+        ExitWithExitCode 1
+    }
+
+    function find-latest-drop($drops) {
+         $result = $null
+         [DateTime]$latest = [DateTime]::New(0)
+         foreach ($drop in $drops) {
+             $dt = [DateTime]::Parse($drop.CreatedDateUtc)
+             if ($result -eq $null -or ($drop.UploadComplete -and !$drop.DeletePending -and ($dt -gt $latest))) {
+                 $result = $drop
+                 $latest = $dt
+             }
+         }
+
+         return $result
+    }
+
+    Write-Host "Acquiring optimization data"
+
+    Create-Directory $IbcOptimizationDataDir
+
+    $dropServiceUrl = "https://devdiv.artifacts.visualstudio.com"
+    # The branh name here needs to be parameterized.
```

Comment:
```suggestion
    # The branch name here needs to be parameterized.
```

(if we go with this before fixing the TODO here :))

Reply:
Fixed